src/main/java/org/radarcns/stream/KafkaStreamFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarcns.stream;
+
+import org.radarcns.config.RadarBackendOptions;
+import org.radarcns.config.RadarPropertyHandler;
+import org.radarcns.stream.empatica.E4StreamMaster;
+import org.radarcns.stream.phone.PhoneStreamMaster;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class KafkaStreamFactory {
+    private final RadarPropertyHandler properties;
+    private final RadarBackendOptions options;
+
+    public KafkaStreamFactory(RadarBackendOptions options,
+                              RadarPropertyHandler properties) {
+        this.options = options;
+        this.properties = properties;
+    }
+
+    public StreamMaster createStreamWorker() throws IOException {
+        String streamType = properties.getRadarProperties().getStreamWorker();
+        if (streamType == null) {
+            // Try to get the stream type from the commandline arguments
+            String[] args = options.getSubCommandArgs();
+            if (args == null || args.length == 0) {
+                streamType = "all";
+            } else {
+                streamType = args[0];
+            }
+        }
+
+        boolean isStandalone = properties.getRadarProperties().isStandalone();
+        switch (streamType) {
+            case "e4":
```

Comment:
I was thinking about something like that
```java
public enum StreamType {
  BIOVOTION("biovotion"),
  EMPATICA("empatica"),
  FITBIT("fitbit"),
  PEBBLE("pebble"),
  PHONE("phone");

  private final String name;

  StreamType(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }
}
```
It is still a `String`, but it can be reused and you can iterate over them.
If we had a static map like
```
static Map<StreamType, StreamMaster> definition = new HasMap<>();

static{
  definition.put(EMPATICA, new E4StreamMaster(isStandalone));
  ...
}
```
we can then remove the `all` case, while in the default we can loop over the enumerator and add all the `StreamMaster` contained in the map.